addons/analytic/models/analytic_plan.py

```diff
@@ -105,6 +106,13 @@ def _compute_children_count(self):
         for plan in self:
             plan.children_count = len(plan.children_ids)
 
+    @api.constrains('parent_id')
+    def _check_different_company_on_parent_plan(self):
+        if not self.parent_id:
+            return
+        if self._origin.parent_id.company_id != self.company_id:
```

Comment:
Why is `_origin` needed? 🧐

Reply:
Can you also write a test and change the commit title since your changes are in `analytic` module and not `project` one? Thanks in advance!